modules/component/src/core/cd-aware/strategy.ts

```diff
@@ -0,0 +1,330 @@
+ * | -------- | -------------------| ---------- | -------------- |
+ * | ZoneFull | cdRef.markForCheck | ❌         | None           |
+ * | ZoneLess | cdRef.markForCheck | ❌         | None           |
+ *
+ * - Ivy
+ *
+ * |          | Render Method      | Coalescing | Coalesce Scope |
+ * | -------- | -------------------| ---------- | -------------- |
+ * | ZoneFull | cdRef.markForCheck | ❌         | None           |
+ * | ZoneLess | cdRef.markForCheck | ❌         | None           |
+ */
+export function createIdleStrategy<T>(
+  cfg: Pick<StrategyFactoryConfig, 'cdRef' | 'component'>
+): CdStrategy<T> {
+  return {
+    render: (): void => {
+      isIvy() ? markDirty(cfg.component) : cfg.cdRef.markForCheck();
+    },
+    behaviour: () => o => o,
+    name: 'idle',
+  };
+}
+
+/**
+ * Noop Strategy
+ *
+ * This strategy is does nothing.
+ *
+ * - \>=ViewEngine
+ * |          | Render Method      | Coalescing | Coalesce Scope |
+ * | -------- | -------------------| ---------- | -------------- |
+ * | ZoneFull | None               | ❌         | None           |
+ * | ZoneLess | None               | ❌         | None           |
+ *
+ * - Ivy
+ *
+ * |          | Render Method      | Coalescing | Coalesce Scope |
+ * | -------- | -------------------| ---------- | -------------- |
+ * | ZoneFull | None               | ❌         | None           |
+ * | ZoneLess | None               | ❌         | None           |
+ */
+export function createNoopStrategy<T>(cfg?: any): CdStrategy<T> {
+  return {
+    render: (): void => {},
+    behaviour: () => o => o,
+    name: 'noop',
+  };
+}
+
+/**
+ *
+ * Pessimistic1 Strategy
+ *
+ * This strategy is providing only Angular built-in features.
+ *  If Ivy is present it uses the latest methods as they are.
+ *
+ * - \>=ViewEngine
+ *
+ * |          | Render Method      | Coalescing | Coalesce Scope |
+ * | -------- | -------------------| ---------- | -------------- |
+ * | ZoneFull | cdRef.markForCheck | ❌         | None           |
+ * | ZoneLess | cdRef.detectChange | ✔️         | None           |
+ *
+ * - Ivy
+ *
+ * |          | Render Method      | Coalescing | Coalesce Scope |
+ * | -------- | -------------------| ---------- | -------------- |
+ * | ZoneFull | ɵmarkDirty         | ❌         | None           |
+ * | ZoneLess | ɵmarkDirty         | ❌         | None           |
+ */
+export function createPessimistic1Strategy<T>(
+  cfg: StrategyFactoryConfig
+): CdStrategy<T> {
+  const inIvy = isIvy();
+  const inZone = hasZone(cfg.ngZone);
+  const durationSelector = getSaveDurationSelector(cfg.ngZone);
+
+  function render() {
+    if (inZone && !inIvy) {
+      cfg.cdRef.markForCheck();
+    } else if (!inZone && !inIvy) {
+      cfg.cdRef.detectChanges();
+    } else {
+      markDirty(cfg.component);
+    }
+  }
+
+  const behaviour = (o$: Observable<T>): Observable<T> => {
+    return !inZone && !inIvy
+      ? o$
+          .pipe
+          // @Notice this part of the code is in the coalescing PR
```

Comment:
I'm not sure what this means.... maybe add a link to the PR here?

Reply:
https://github.com/ngrx/platform/pull/2456
Will also add it to the code.

Reply:
Done